src/go/k8s/api/redpanda/v1alpha1/redpanda_clusterspec_types.go

```diff
@@ -621,6 +624,12 @@ type Statefulset struct {
 	TerminationGracePeriodSeconds *int `json:"terminationGracePeriodSeconds,omitempty"`
 }
 
+// PodTemplate will pass label and annotation to Statefulset Pod template.
+type PodTemplate struct {
```

Comment:
I find this to be confusing because there is a `corev1.PodTemplate` type that already exists that is _very_ similar to this type but not fully compatible. Could we use `corev1.PodTemplate` here, make this type compatible with `corev1.PodTemplate` by placing a top level `metav1.ObjectMeta` into this struct, or rename this struct to not clash with Kubernetes' built in types?

Reply:
> Could we use corev1.PodTemplate here, make this type compatible with corev1.PodTemplate by placing a top level metav1.ObjectMeta into this struct

I'm not sure I follow.

Do you want to get rid of `type PodTemplate struct` and use `corev1.PodTemplate` in https://github.com/redpanda-data/redpanda-operator/pull/117/files#diff-177b16be1e602cf0430b5fa8ae9b3027794c498779b3652c1038551a9aa074b8R584

Or You would like to see 
```go
type PodTemplate struct {
        metav1.ObjectMeta
	Labels      map[string]string `json:"labels,omitempty"`
	Annotations map[string]string `json:"annotations,omitempty"`
}
```

> or rename this struct to not clash with Kubernetes' built in types?

I can rename the struct, but the values.yaml has already released [podTemplate](https://github.com/redpanda-data/helm-charts/blob/a2443fbe4bd1620af2d45dd82b697398c36a512c/charts/redpanda/values.yaml#L633-L637).

Reply:
Ah, sorry. I missed that we have a `podTemplate` in values.yaml. Thank you for the link! Let's keep it as is for now. Maybe we can add a TODO or file an issue to make this struct match `corev1.PodTemplate` in the future?